examples/examples-cpu/nyc-taxi/xgboost-dask.ipynb

```diff
@@ -192,25 +203,73 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "import dask_xgboost\n",
-    "\n",
-    "xgb_reg = dask_xgboost.XGBRegressor(\n",
-    "    objective=\"reg:squarederror\",\n",
-    "    tree_method='approx',\n",
-    "    learning_rate=0.1,\n",
-    "    max_depth=5,\n",
-    "    n_estimators=50,\n",
+    "import xgboost as xgb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Training data for `xgboost.dask` needs to be prepared in a special object called `DaskDMatrix`. This is like the XGBoost `DMatrix` that you might be familiar with, but is backed by Dasks's distributed collections (Dask DataFrame and Dask Array)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dtrain = xgb.dask.DaskDMatrix(\n",
+    "    client=client,\n",
+    "    data=taxi_train[features],\n",
+    "    label=taxi_train[y_col]\n",
     ")"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can pass any [xgboost parameters](https://xgboost.readthedocs.io/en/latest/parameter.html) to `xgb.dask.train()`. The training process will then start up on all workers that have some of the data in `dtrain`."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "%%time\n",
-    "_ = xgb_reg.fit(taxi_train[features], y=taxi_train[y_col])"
+    "result = xgb.dask.train(\n",
```

Comment:
I noticed that the old dask-xgboost examples (and single-node xgboost examples) use the scikit-learn API. Is there a reason to use the functional API here over the scikit-learn API? Should we favor consistency across all our examples, or do we want to show both functional and scikit-learn APIs to cater to a variety of users?

Reply:
no strong preference, I'll change it to the scikit-learn API

Reply:
ok @rikturr actually, can I just leave this with the functional API?

My most immediate reason is that I just tried using the scikit-learn interface and got a pickling error that I think will take a bit of time to debug and figure out

![image](https://user-images.githubusercontent.com/7608904/102412492-2a883000-3fb9-11eb-9537-c6103da61f2b.png)

Other reasons I like the functional API:

1. you pass in a dictionary of XGBoost parameters, instead of needing to match to keyword args (which are not always the same names as the lower-level XGBoost parameters)
2. https://www.saturncloud.io/docs/tutorials/xgboost/ already uses the functional API (because of number 1), so this would make the examples consistent with that

Reply:
Ok no problem, LGTM!